src/Server/TCPHandler.cpp

```diff
@@ -501,6 +501,7 @@ void TCPHandler::runImpl()
             /// (i.e. deallocations from the Aggregator with two-level aggregation)
             state.reset();
             query_scope.reset();
+            last_sent_snapshots.clear();
```

Comment:
`unordered_map::clear` is not required to release memory. Also, this deallocation should be done before `query_scope.reset()`.

Reply:
That diff is not about memory. It has to be cleaned otherwise wrong metrics are send to the client. That line is about `fix negative counters`.

> his deallocation should be done before query_scope.reset()

Why? I do not catch.